File: Day27/count-hills-and-valleys-in-an-array.js
// Time: O(N)  ::  Space: O(1)
var countHillValley = function(nums) {
    let count = 0;
    let temp = 1;

    for (let i = 1; i < nums.length - 1; i++) {
        if (nums[i] === nums[i + 1]) {
            temp++;
        } else if ((nums[i - temp] > nums[i] && nums[i] < nums[i + 1]) ||
                   (nums[i - temp] < nums[i] && nums[i] > nums[i + 1])) {
            temp = 1;
            count++;
        } else {
            temp = 1;
        }
    }
    return count;
};

File: Day27/count-hills-and-valleys-in-an-array.py
# Time: O(N)  ::  Space: O(1)
class Solution:
    def countHillValley(self, nums: List[int]) -> int:
        count =0
        temp = 1
        i = 1

        for i in range(1, len(nums)-1):
            if nums[i] == nums[i+1]:
                temp+=1
            elif nums[i-temp] > nums[i] <nums[i+1] or nums[i-temp] < nums[i] >nums[i+1]:
                temp = 1
                count+=1
            else:
                temp = 1
        return count
        